Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPISidebarCocoa.mm

```
@@ -266,29 +302,28 @@
         return;
     }
 
-    auto maybeSidebar = getOrCreateSidebar(*window);
+    if (!hasActiveUserGesture(*tab)) {
+        completionHandler(toWebExtensionError(nil, nil, @"toggle can only be called in response to a user action"));
+        return;
+    }
+
+    auto maybeSidebar = getOrCreateSidebar(*tab);
     if (!maybeSidebar) {
         completionHandler(toWebExtensionError(nil, nil, @"could not create sidebar"));
         return;
     }
 
-    auto& sidebar = maybeSidebar.value();
-    if (sidebar->isOpen()) {
-        if (!sidebar->canProgrammaticallyCloseSidebar()) {
-            completionHandler(toWebExtensionError(nil, nil, @"it is not implemented"));
-            return;
-        }
-
-        closeSidebarForTab(*tab);
-    } else {
-        if (!sidebar->canProgrammaticallyOpenSidebar()) {
-            completionHandler(toWebExtensionError(nil, nil, @"it is not implemented"));
-            return;
-        }
-
-        openSidebarForTab(*tab);
+    if (!canProgrammaticallyCloseSidebar() || !canProgrammaticallyOpenSidebar()) {
+        completionHandler(toWebExtensionError(nil, nil, @"it is not implemented"));
+        return;
     }
```

Should this be earlier, so it fails faster?

---

Also add API name for first parameter.